internal/scti/ctlog.go

```diff
@@ -43,10 +42,18 @@ type Storage interface {
 	GetCertDedupInfo(context.Context, *x509.Certificate) (dedup.SCTDedupInfo, bool, error)
 }
 
+// ChainValidator provides functions to validate incoming chains.
+type ChainValidator interface {
+	Validate(req rfc6962.AddChainRequest, expectingPrecert bool) ([]*x509.Certificate, error)
+	Roots() []*x509.Certificate
+}
```

Comment:
There is only one chain validator implementation. Do we really need this interface?

Reply:
Yeah - I was expecting this comment, I thought the same when I opened this PR a days after I wrote it now. It's mostly stylistic, I got quite excited when I saw that a log struct is just a collection of functions {signSCT, storage funcs, root validation funcs}. I could just make the ChainValidator object public, and use this one directly. I don't feel strongly about it.

Reply:
Okay. Let's get this in.